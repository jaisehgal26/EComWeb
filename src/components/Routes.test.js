// import React from "react";
// import { Route } from "react-router";
// import { shallow, mount } from "enzyme";
// import toJson from "enzyme-to-json";
// import Routes from "./Routes";
// import Home from "./Home";
// import Login from "./Login";
// import Cart from "./Cart";
// import Products from "./Products";

// test("should render correctly", () => {
//   const wrapper = toJson(shallow(<Routes />));
//   expect(wrapper).toMatchSnapshot();
// });

// test("renders correct route home", () => {
//   const wrapper = shallow(<Routes />);
//   console.log(wrapper);
//   const pathMap = wrapper.find(Route).reduce((pathMap, route) => {
//     const routeProps = route.props();
//     pathMap[routeProps.path] = routeProps.component;
//     return pathMap;
//   }, {});

//   expect(pathMap["/"]).toEqual(<Home />);
// });

// // test("renders correct route login", () => {
// //   const wrapper = shallow(<Routes />);
// //   const pathMap = wrapper.find(Route).reduce((pathMap, route) => {
// //     const routeProps = route.props();
// //     if (routeProps.component) {
// //       pathMap[routeProps.path] = routeProps.component;
// //     } else if (routeProps.render) {
// //       pathMap[routeProps.path] = routeProps.render({}).type;
// //     }
// //     return pathMap;
// //   }, {});

// //   expect(pathMap["/login"]).toEqual(<Login />);
// // });
// // test("renders correct route Cart", () => {
// //   const wrapper = shallow(<Routes />);
// //   const pathMap = wrapper.find(Route).reduce((pathMap, route) => {
// //     const routeProps = route.props();
// //     pathMap[routeProps.path] = routeProps.component;
// //     return pathMap;
// //   }, {});

// //   expect(pathMap["/Cart"]).toBe(Cart);
// // });
