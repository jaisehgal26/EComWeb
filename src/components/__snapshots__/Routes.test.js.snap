// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<BrowserRouter>
  <Suspense
    fallback={
      <div
        className="center spinner-grow text-warning"
        role="status"
      >
        <span
          className="sr-only"
        >
          Loading...
        </span>
      </div>
    }
  >
    <Routes>
      <Route
        element={<Home />}
        exact={true}
        path="/"
      />
      <Route
        element={<Login />}
        path="/login"
      />
      <Route
        element={<Cart />}
        path="/cart"
      />
      <Route
        element={<Products />}
        path="/products"
      />
      <Route
        element={<AddProduct />}
        path="/products/addproduct"
      />
    </Routes>
  </Suspense>
</BrowserRouter>
`;
